tensorflow/java/src/main/native/saved_model_bundle_jni.cc

```diff
@@ -21,19 +21,40 @@ limitations under the License.
 #include "tensorflow/java/src/main/native/saved_model_bundle_jni.h"
 
 namespace {
+void TF_MaybeDeleteBuffer(TF_Buffer* buf) {
+  if (buf == nullptr) return;
+  TF_DeleteBuffer(buf);
+}
+
+typedef std::unique_ptr<TF_Buffer, decltype(&TF_MaybeDeleteBuffer)>
+    unique_tf_buffer;
 
+unique_tf_buffer MakeUniqueBuffer(TF_Buffer* buf) {
+  return unique_tf_buffer(buf, TF_MaybeDeleteBuffer);
+}
 }  // namespace
+
 JNIEXPORT jobject JNICALL Java_org_tensorflow_SavedModelBundle_load(JNIEnv * env,
                                                                     jclass clazz,
                                                                     jstring export_dir,
-                                                                    jobjectArray tags) {
+                                                                    jobjectArray tags,
+                                                                    jbyteArray run_options) {
 
   TF_Status* status = TF_NewStatus();
   jobject bundle = nullptr;
 
   // allocate parameters for TF_LoadSessionFromSavedModel
   TF_SessionOptions* opts = TF_NewSessionOptions();
-  const TF_Buffer* crun_options = nullptr;
+  unique_tf_buffer crun_options(MakeUniqueBuffer(nullptr));
```

Comment:
Can we do away with all this MakeUniqueBuffer fanciness and instead have:

```
TF_Buffer* crun_options = nullptr;
// ...
if (crun_options != nullptr) TF_DeleteBuffer(crun_options);
```

Since there is only one place where we create the TF_Buffer and only one path to the point where it can be deleted, just like all other cleanup functions?